src/lib/items/RouteData.php

```diff
@@ -7,50 +7,352 @@
 
 namespace cebe\yii2openapi\lib\items;
 
+use cebe\openapi\spec\PathItem;
+use cebe\yii2openapi\lib\SchemaResponseResolver;
 use yii\base\BaseObject;
+use yii\base\InvalidCallException;
+use yii\helpers\ArrayHelper;
+use yii\helpers\Inflector;
+use function array_filter;
+use function count;
+use function in_array;
+use function preg_match;
+use function reset;
 
 /**
- * @property-read string $route
- * @property-read string $optionsRoute
+ * @property-read array        $partsWithoutParams
+ * @property-read array        $firstParam
+ * @property-read array        $parts
+ * @property-read string       $lastPart
+ * @property-read  string      $path
+ * @property-read  string      $type
+ * @property-read  string      $pattern
+ * @property-read  string      $controller
+ * @property-read  string      $action
+ * @property-read  string|null $idParam
+ * @property-read  string|null $parentParam
+ * @property-read  array       $params
  */
-class RouteData extends BaseObject
+final class RouteData extends BaseObject
 {
-    /**@var string */
-    public $path;
+    private const PATTERN_PROFILE = '~^/(?<controller>me)$~';
+    /**
+     * @example /auth/login
+     * @example /auth/reset/password
+     * @example /auth/verify/{email}
+     */
+    private const PATTERN_ACTION = '~^/(?<controller>[\w-]+)/(?<action>[\w-]+)/?(?<method>[\w-]+)?/?(?:[{}/\w-]+)?$~';
+    /** @example /posts */
+    private const PATTERN_LIST = '~^/(?<controller>[\w-]+)$~';
+    /**
+     * @example /posts/1
+     * @example /article/4
+     */
+    private const PATTERN_RESOURCE = '~^/(?<controller>[\w-]+)/{(?<idParam>[\w-]+)}$~';
+    /** @example /posts/1/upload/avatar */
+    /** @example /users/1/generate/password */
+    private const PATTERN_RESOURCE_OPERATION
+        = '~^/(?<controller>[\w-]+)/{(?<idParam>[\w-]+)}/(?<action>(?!relationships)[\w-]+)/(?<method>[\w-]+)$~';
+    /**
+     * @example /posts/1/comments
+     * @example /article/4/category
+     */
+    private const PATTERN_LIST_FOR_RESOURCE = '~^/(?<for>[\w-]+)/{(?<parentParam>[\w-]+)}/(?<controller>[\w-]+)$~';
+    /** @example /posts/1/comments/33 */
+    private const PATTERN_FOR_RESOURCE = '~^/(?<for>[\w-]+)/{(?<parentParam>[\w-]+)}/(?<controller>[\w-]+)/{(?<idParam>[\w-]+)}$~';
```

Comment:
An important note about routes for related resources, like /posts/1/comments.
In general, there are two ways to get it.
Via relation : $post = Post::findOne(1); $post->comments;
Or via parent attribute: Comments::where([ 'post_id'=> 1])->all();
Currently, yii2-fractal provides the second variant. (The first variant also can be added if need)
So we need to know the foreign attribute. So we should define path routes like /posts/{postId}/comments - because it is the easiest way to resolving it. (Otherwise, we will need to calculate it from relation object, or try to find another way to define it in openapi schema)

Reply:
> The first variant also can be added if need

Syntax like `$post->comments` is nice in context of DB because query is hidden. In case when API is involved I think it is better to see query explicitly, because API request is more expensive than DB query.

> So we should define path routes like `/posts/{postId}/comments`

it is fine to have conventions here, should be documented so it is well known how it works.
Should we make conversion from `postId` to `post_id` or should we name parameter exactly like the field? Right now I see no need to make conversion.

Reply:
`{postId}` will be lowercased to post_id for attribute 
https://github.com/Insolita/yii2-openapi/blob/json-api-integration/src/lib/items/FractalAction.php#L149